packages/web-components/src/components/masthead/cloud/cloud-masthead-container.ts

```diff
@@ -0,0 +1,98 @@
+import { ActionCreatorsMapObject, Dispatch, Store, bindActionCreators } from 'redux';
+import { customElement } from 'lit-element';
+import Cookies from 'js-cookie';
+import store from '../../../internal/vendor/@carbon/ibmdotcom-services-store/store';
+import { LocaleAPIActions } from '../../../internal/vendor/@carbon/ibmdotcom-services-store/actions/localeAPI.d';
+import { TranslateAPIActions } from '../../../internal/vendor/@carbon/ibmdotcom-services-store/actions/translateAPI.d';
+import { SearchAPIActions } from '../../../internal/vendor/@carbon/ibmdotcom-services-store/actions/searchAPI.d';
+import ConnectMixin from '../../../globals/mixins/connect';
+import { loadTranslation } from '../../../internal/vendor/@carbon/ibmdotcom-services-store/actions/translateAPI';
+import { loadLanguage, setLanguage } from '../../../internal/vendor/@carbon/ibmdotcom-services-store/actions/localeAPI';
+import { loadSearchResults } from '../../../internal/vendor/@carbon/ibmdotcom-services-store/actions/searchAPI';
+import { MastheadContainerState, MastheadContainerStateProps, MastheadContainerActions } from '../masthead-container';
+import DDSCloudMastheadComposite from './cloud-masthead-composite';
+
+const { stablePrefix: ddsPrefix } = ddsSettings;
+
+/**
+ * The cookie name for determining user login status for cloud.ibm.com.
+ */
+const loginStatusCookieName = 'com.ibm.cloud.iam.LoggedIn.manual';
+
+/**
+ * Returns user login status (authenticated or anonymous)
+ */
+function getUserLoginStatus() {
+  return Cookies.get(loginStatusCookieName) === '1' ? 'authenticated' : 'anonymous';
+}
+
+/**
+ * @param state The Redux state for masthead.
+ * @returns The converted version of the given state, tailored for `<dds-masthead-container>`.
+ */
+export function mapStateToProps(state: MastheadContainerState): MastheadContainerStateProps {
+  const { localeAPI, translateAPI, searchAPI } = state;
+  const { language } = localeAPI ?? {};
+  const { translations } = translateAPI ?? {};
+  const { currentSearchQueryString, searchResults } = searchAPI ?? {};
+  let currentSearchResults;
+  for (let { length = 0 } = currentSearchQueryString ?? {}; !currentSearchResults && length > 0; --length) {
+    currentSearchResults = searchResults?.[currentSearchQueryString!.substr(0, length)]?.[language!];
+  }
+  return pickBy(
+    {
+      authenticatedProfileItems: !language ? undefined : translations?.[language]?.masthead?.profileMenu.signedin.links,
+      authenticatedCtaButtons: !language ? undefined : translations?.[language]?.masthead?.profileMenu.signedin.ctaButtons,
+      navLinks: !language ? undefined : translations?.[language]?.mastheadNav?.links,
+      unauthenticatedProfileItems: !language ? undefined : translations?.[language]?.masthead?.profileMenu.signedout.links,
+      unauthenticatedCtaButtons: !language ? undefined : translations?.[language]?.masthead?.profileMenu.signedout.ctaButtons,
```

Comment:
Since the `ctaButtons` key is new, I think we will have to add this to the `packages/services-store/src/types/translateAPI.ts` file